crates/distances/src/sets/mod.rs

```diff
@@ -87,3 +87,46 @@ pub fn dice<T: Int, U: Float>(x: &[T], y: &[T]) -> U {
 
     U::one() - (U::from(2) * intersection_size / size)
 }
+
+/// Kulsinski distance.
+///
+/// Similar to the Jaccard distance, the Kulsinski distance is a measure of the dissimilarity
+/// between two sets. It is defined as the cardinality of the intersection divided by the
+/// sum of the number of not equal dimensions and the total number of dimensions.
```

Comment:
Please add a link to your source for the definition of Kulsinski distance between merging. There are some different definitions out there. For example, [this definition from Scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.kulczynski1.html#scipy.spatial.distance.kulczynski1) is different from what you have. [This one](https://docs.scipy.org/doc/scipy-1.7.1/reference/reference/generated/scipy.spatial.distance.kulsinski.html) is an equivalent definition to what you have, but it is in terms of Boolean vectors, and it takes a bit of algebra to verify they are equivalent.